modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -538,12 +538,25 @@ else if (Objects.equals(
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
-			if (!(objectValidationRuleSetting.compareName(
+			if (FeatureFlagManagerUtil.isEnabled("LPS-187854") &&
+				!(objectValidationRuleSetting.compareName(
 					ObjectValidationRuleSettingConstants.
```

Comment:
Do we need to also check NAME_KEY_OBJECT_FIELD_ID here? It seems that it's already covered in the if condition below. We should keep NAME_KEY_OBJECT_FIELD_ID working as before, even without this feature flag.